src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/StorageServices/NewAzureVhdVMCommand.cs

```diff
@@ -0,0 +1,576 @@
+            });
+        }
+
+        private void InitializeOptionalParameters()
+        {
+            if (string.IsNullOrEmpty(this.VMName))
+            {
+                this.VMName = string.Format("vm-{0}{1}{2}{3}", this.Location, DateTime.Now.ToString("yyyyMMddhhmmss"), this.ResourceGroupName, Path.GetRandomFileName().Replace(".", ""));
+            }
+
+            if (string.IsNullOrEmpty(this.VMSize))
+            {
+                this.VMSize = "Standard_A2";
+            }
+
+            if (this.SecurityRules == null)
+            {
+                this.SecurityRules = new List<SecurityRule>()
+                {
+                    new SecurityRule
+                    {
+                        Name = "allow3389",
+                        Description = "allow3389",
+                        Access = "Allow",
+                        Protocol = "Tcp",
+                        Direction = "Inbound",
+                        Priority = 110,
+                        SourceAddressPrefix = "*",
+                        SourcePortRange = "*",
+                        DestinationAddressPrefix = "*",
+                        DestinationPortRange = "3389"
+                    },
+                    new SecurityRule
+                    {
+                        Name = "allow80",
+                        Description = "allow80",
+                        Access = "Allow",
+                        Protocol = "Tcp",
+                        Direction = "Inbound",
+                        Priority = 120,
+                        SourceAddressPrefix = "*",
+                        SourcePortRange = "*",
+                        DestinationAddressPrefix = "*",
+                        DestinationPortRange = "80"
+                    },
+                    new SecurityRule
+                    {
+                        Name = "allow8080",
+                        Description = "allow8080",
+                        Access = "Allow",
+                        Protocol = "Tcp",
+                        Direction = "Inbound",
+                        Priority = 130,
+                        SourceAddressPrefix = "*",
+                        SourcePortRange = "*",
+                        DestinationAddressPrefix = "*",
+                        DestinationPortRange = "8080"
+                    }
+                };
+            }
+
+            if (string.IsNullOrEmpty(this.DefaultVNetAddressSpace))
+            {
+                this.DefaultVNetAddressSpace = "10.0.0.0/16";
+            }
+
+            if (string.IsNullOrEmpty(this.DefaultSubnetAddressSpace))
+            {
+                this.DefaultSubnetAddressSpace = "10.0.0.0/24";
+            }
+        }
+
+        private void ValidateInitialInputParameters()
+        {
+            if (this.DiskFile == null && this.DiskLink == null)
+            {
+                throw new ArgumentException("DiskFile and DiskLink cannot be both null.");
+            }
```

Comment:
Do you really need this condition?  DiskFile and DiskLink are in different parameter sets, and there are already notnull validator in the parameters.

Reply:
Just to be safe...

Reply:
This is unnecessary, remove